processor/processor.go

```diff
@@ -0,0 +1,114 @@
+package processor
+
+import (
+	"fmt"
+	"path"
+
+	"github.com/algorand/go-algorand/agreement"
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/bookkeeping"
+	"github.com/algorand/go-algorand/ledger"
+	"github.com/algorand/go-algorand/ledger/ledgercore"
+	"github.com/algorand/go-algorand/logging"
+	"github.com/algorand/go-algorand/rpcs"
+)
+
+// Processor is the block processor interface
+type Processor interface {
+	Process(cert *rpcs.EncodedBlockCert) error
+	SetHandler(handler func(block *ledgercore.ValidatedBlock) error)
+	NextRoundToProcess() uint64
+	Init(path string, genesis *bookkeeping.Genesis, genesisBlock *bookkeeping.Block) error
+}
+
+type processorImpl struct {
```

Comment:
Could you rename this to `blockProcessor` and move the implementation into a new package?